src/util/TimeUtil.js

```diff
@@ -0,0 +1,93 @@
+define(['parse-ms', '@sindresorhus/to-milliseconds'], function (parseMs, toMilliseconds) {
+
+  const MOMENT_UNIT = {
+    MILLISECOND: 'milliseconds',
+    SECOND: 'seconds',
+    MINUTE: 'minutes',
+    HOUR: 'hours',
+    DAY: 'days'
+  };
+
+  const MOMENT_UNIT_KEYS = Object.keys(MOMENT_UNIT);
+
+  /**
+   * @method convertMomentUnits
+   * Conversion between moment's units and our units internally
+   *
+   * @param {String} momentUnit The units that val is in
+   * @return {String} The key in the MOMENT_UNIT hash
+   */
+  const convertMomentUnits = function (momentUnit) {
+    const entry = MOMENT_UNIT_KEYS.filter(k => MOMENT_UNIT[k] === momentUnit);
+
+    return entry.length === 1 ? entry[0] : momentUnit;
+  };
+
+  return {
+
+    /**
+     * @method getTimeInHighestRelevantUnit
+     * Will return a number in the units of the highest relevant time unit.
+     * Only checks milliseconds, seconds, minutes, hours, and days.
+     * E.g.
+     *   15 minutes -> 15 minutes
+     *   60 minutes ->  1 hours
+     *   90 minutes -> 90 minutes
+     *   24 hours   ->  1 days
```

Comment:
Would this return "1 hours" and "1 days"?

Reply:
Synced up with Haisheng; for the purposes of this PR, it's fine, as it's existing problematic functionality with TimeUtil from Courage

Reply:
Actually the description is not accurate.
The unit in the return object is canonical.
Updated comments.